bluesky/plans.py

```diff
@@ -283,18 +301,36 @@ def log_scan(detectors, motor, start, stop, num, *, per_step=None, md=None):
     See Also
     --------
     :func:`bluesky.plans.rel_log_scan`
+    :func:`bluesky.plans.log_grid_scan`
+    :func:`bluesky.plans.rel_log_grid_scan`
     """
+
+    if len(args) % 3 != 0:
+        raise ValueError('The list of arguments handed to ``log_scan()`` must '
+                         'contain multiples of the 3 arguments `motor`, '
+                         '`start`, `stop`. However what was recieved is of '
+                         'length {len(args)} which is not divisble by 3.'
+                         'The full set of arguments are {}.'.format(len(args),
+                                                                    args))
+
+    list_args = []
+    repr_args = []
+    for motor, start, stop in partition(3, args):
+        if start <= 0 or stop <= 0:  # resolve 'math.log' error if necesary
+            offset = -1*min(start, stop)+1
+        else:
+            offset = 0
+
+        list_args.extend([motor, np.logspace(math.log(start+offset, base),
```

Comment:
I think `offset` is not necessary and this should be simply `np.logspace(start, stop, num, base=base`)`, it was before. This change seems to be an API break unless I am misunderstanding.

From `numpy.logspace`:
```
Parameters
----------
start : array_like
    ``base ** start`` is the starting value of the sequence.
stop : array_like
    ``base ** stop`` is the final value of the sequence, unless `endpoint`
    is False.  In that case, ``num + 1`` values are spaced over the
    interval in log-space, of which all but the last (a sequence of
    length `num`) are returned.
```

Reply:
@danielballan I should have documented this correctly.

I don't think this is an API break as per the current documentation below start and stop are the start postion and strop position, as they are for all other plans we have:
>def log_scan(detectors, motor, start, stop, num, *, per_step=None, md=None):
    """
    Scan over one variable in log-spaced steps.
    Parameters
    ----------
    detectors : list
        list of 'readable' objects
    motor : object
        any 'settable' object (motor, temp controller, etc.)
    start : float
        starting position of motor
    stop : float
        ending position of motor
    num : int
        number of steps
    per_step : callable, optional
        hook for customizing action of inner loop (messages per step)
        Expected signature: ``f(detectors, motor, step)``
    md : dict, optional
        metadata
    See Also
    --------
    :func:`bluesky.plans.rel_log_scan`
    """

Adding `offset` in here allows for <=0 start  and/or stop positions, it is set to zero for all other cases. So this is an extension not an API break.

I am however happy for this to wait until detailed review.

Reply:
I agree offset is an extension. Focusing on the other part, I think it is possible to interpret changing start and stop as either:

* a bug fix — it changes the behavior to match the documentation and to be consistent with the other plans, or
* an API change — typing the same arguments now moves the motors to a different place than before.

I mostly agree that this PR would be an improvement, but it does seem a potentially dangerous to change a plan in a way that causes it to put the motors in very different positions than it did before. Anyone who is using it might be caught by surprise by the change.

Radical proposal: if we agree that the current implementation is not good, we might consider addressing that by deprecating log_scan entirely rather than fixing it in a way that could break user code in a surprising way. We agree that usage is rare, possibly nonexistent. Maybe it’s better to let the individual beamline scientist choose the parametrization that makes sense to their users by way of a custom plan that wraps list_scan and/or list_grid_scan.

Reply:
Thanks for the feedback @danielballan 

>an API change — typing the same arguments now moves the motors to a different place than before.

I am just now realizing that the current `log_scan` behaviour is not consistent with it's docstring. I agree that this could be then viewed as an API change, and that it is probably a bad move.

>Radical proposal: if we agree that the current implementation is not good, we might consider addressing that by deprecating log_scan entirely rather than fixing it in a way that could break user code in a surprising way.

As a counter-proposal how about we rename the scans in this PR as 'logstep_scan' , ....
This has the benefits:
- it removes the potential API break, and introduces the API consistent with existing plans.
- it's name matches more closely to it's purpose (to scan in log-spaced steps over one or more axis/axes).

We could then consider deprecating `log_scan` long term.

Reply:
I don't have strong feelings either way, but I would lean toward not adding two new plans if we think that are rarely/never used. Might be worth pulling this back up to a GH issue for wider consideration.